src/app/renderer.js

```diff
@@ -8,17 +8,22 @@ var utils = require('./utils');
 function Renderer (editor, compiler, updateFiles) {
 
   var detailsOpen = {};
-  var executionContext = 'vm';
+  var executionContext = web3.injected ? 'injected' : 'vm';
```

Comment:
It doesn't fail. From udapp's perspective, it's either using the vm object or a web3 object. If the `executationContext` isn't `vm`, then it uses the a web3 object. That object is either setup manually with a provider, or gets the provider injected in `web3-adapter.js`.